src/transformers/models/mgp_str/processing_mgp_str.py

```diff
@@ -63,41 +78,81 @@ def __init__(self, image_processor=None, tokenizer=None, **kwargs):
                 FutureWarning,
             )
             feature_extractor = kwargs.pop("feature_extractor")
+        if "char_tokenizer" in kwargs:
+            warnings.warn(
+                "The `char_tokenizer` argument is deprecated and will be removed in future versions, use `tokenizer`"
+                " instead.",
+                FutureWarning,
+            )
+            char_tokenizer = kwargs.pop("char_tokenizer")
 
         image_processor = image_processor if image_processor is not None else feature_extractor
+        tokenizer = tokenizer if tokenizer is not None else char_tokenizer
         if image_processor is None:
             raise ValueError("You need to specify an `image_processor`.")
         if tokenizer is None:
             raise ValueError("You need to specify a `tokenizer`.")
 
-        self.char_tokenizer = tokenizer
+        self.tokenizer = tokenizer
+        self.char_tokenizer = tokenizer  # For backwards compatibility
```

Comment:
This should be enough, with a deprecation warning whenever one tries to access or re-set with `processor.chat_tokenizer`

Reply:
I've converted it to a property instead so we can raise a warning whenever it's used/set